src/millie.nim

```diff
@@ -10,46 +10,41 @@ const units = @[
   "P", # quadrillion
 ]
 
-## Converts big numbers to what's pleasant to see (an adorable, little girl, perhaps?)
+
 proc millify*(
     value: int64,
     precision: int = 1,
     lowercase: bool = false,
     space: bool = false,
   ): string =
+  ## Converts big numbers to what's pleasant to see (an adorable, little girl, perhaps?)
   
-  # validate if `vaue` can be parsed or not
-  var val: SomeNumber = parseNum(value)
+  # preallocate the resulting string
+  result = newStringOfCap(10)
 
-  # `prefix` will be used later when assembling
-  let prefix: string = if val < 0: "-" else: ""
+  var val = float(value)
```

Comment:
Nim is a statically typed language - so things like `parseNum` aren't needed at all.
If we just want to convert the value from int64 to float we can use `float` like here